### ISSUE TITLE
Rename default branch from master to main

### ISSUE THREAD

**Issue by sophie-h** [gtk-rs/release#139]
Posting this in release since I think all repos should be changed.

Maybe do this before the hackfest such that we can fix everything that is affected by this during the weekend?

**Comment by sdroege** [comment]
As a first step this needs some documenting of the impact of this. Apart from switching the default branch in git, what is assuming a branch named "master"?

Probably the CI scripts, probably @GuillaumeGomez's release scripts. What else?

Also how do we make sure that people actually notice this change and we don't get bug reports in a few months from people running from a completely outdated master branch?

**Comment by sophie-h** [comment]
> Also how do we make sure that people actually notice this change and we don't get bug reports in a few months from people running from a completely outdated master branch?

https://github.com/github/renaming/ sounds like github got us covered for many things.

Some background information https://about.gitlab.com/blog/2021/03/10/new-git-default-branch-name/. Looks like main will be the default in the majority of place pretty soon. Maybe it's less pain to do the change now.

**Comment by GuillaumeGomez** [comment]
The default branch for cargo is `master`. Another thing to be taken into account.

**Comment by sophie-h** [comment]
> The default branch for cargo is `master`. Another thing to be taken into account (for our end users).

The branch for `cargo new` depends on `init.defaultBranch`. Support for using the remotes default branch is only waiting for a release I guess https://github.com/rust-lang/cargo/blob/master/CHANGELOG.md#cargo-152-2021-05-06.

**Comment by GuillaumeGomez** [comment]
Not for new projects, for projects using `gtk-rs` as a dependency.

**Comment by sophie-h** [comment]
> Not for new projects, for projects using `gtk-rs` as a dependency.

Yes I answered this in my comment too. Also, this change will probably be transparent for end users because GitHub redirects the branch  https://about.gitlab.com/blog/2021/03/10/new-git-default-branch-name/ (also mentioned before.)

**Comment by GuillaumeGomez** [comment]
Oh indeed, well, that's one less concern then.

**Comment by sophie-h** [comment]
Progress with Rust 1.53

> Cargo no longer assumes the default HEAD of git repositories is named master. This means you no longer need to specify branch = "main" for git dependencies from a repository where the default branch is called main.

https://blog.rust-lang.org/2021/06/17/Rust-1.53.0.html#head-branch-name-support-in-cargo

**Comment by sdroege** [comment]
Should we then maybe do it before the release now? :) @GuillaumeGomez @bilelmoussaoui ?

**Comment by bilelmoussaoui** [comment]
Sounds good to me
